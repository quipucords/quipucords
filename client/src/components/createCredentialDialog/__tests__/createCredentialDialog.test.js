import React from 'react';
import configureMockStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import CreateCredentialDialog from '../createCredentialDialog';

describe('CreateCredentialDialog Component', () => {
  const generateEmptyStore = () => configureMockStore()({ credentials: {}, viewOptions: {} });

  it('should shallow render a basic component', () => {
    const store = generateEmptyStore();
    const props = { show: true };
    const wrapper = shallow(<CreateCredentialDialog {...props} />, { context: { store } });

    expect(wrapper.dive()).toMatchSnapshot();
  });
});
