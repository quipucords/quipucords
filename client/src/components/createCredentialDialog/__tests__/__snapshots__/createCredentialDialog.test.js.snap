// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCredentialDialog Component should shallow render a basic component 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={false}
    closeLabel="Close"
  >
    <Button
      active={false}
      aria-hidden="true"
      aria-label="Close"
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="close"
      disabled={false}
      onClick={[Function]}
    >
      <Icon
        name="close"
        type="pf"
      />
    </Button>
    <ModalTitle
      bsClass="modal-title"
      componentClass="h4"
    >
      Add Credential
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  />
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={true}
  >
    <Form
      bsClass="form"
      componentClass="form"
      horizontal={true}
      inline={false}
    >
      <FormGroup
        bsClass="form-group"
      >
        <Col
          bsClass="col"
          componentClass={[Function]}
          sm={5}
        >
          Source Type
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          sm={7}
        >
          <FormControl
            bsClass="form-control"
            className="quipucords-form-control"
            componentClass="input"
            readOnly={true}
            type="text"
            value=""
          />
        </Col>
      </FormGroup>
      <FormGroup
        bsClass="form-group"
        validationState={null}
      >
        <Col
          bsClass="col"
          componentClass={[Function]}
          sm={5}
        >
          Credential Name
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          sm={7}
        >
          <FormControl
            autoFocus={true}
            bsClass="form-control"
            className="quipucords-form-control"
            componentClass="input"
            onChange={[Function]}
            placeholder="Enter a name for the credential"
            type="text"
            value=""
          />
        </Col>
      </FormGroup>
      <FormGroup
        bsClass="form-group"
      >
        <Col
          bsClass="col"
          componentClass={[Function]}
          sm={5}
        >
          Authentication Type
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          sm={7}
        >
          <DropdownSelect
            className=""
            id="auth-type-select"
            multiselect={false}
            onSelect={[Function]}
            options={
              Object {
                "sshKey": "SSH Key",
                "usernamePassword": "Username and Password",
              }
            }
            pullRight={false}
            selectValue="usernamePassword"
            title="Select option"
          />
        </Col>
      </FormGroup>
      <FormGroup
        bsClass="form-group"
        validationState={null}
      >
        <Col
          bsClass="col"
          componentClass={[Function]}
          sm={5}
        >
          Username
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          sm={7}
        >
          <FormControl
            bsClass="form-control"
            componentClass="input"
            onChange={[Function]}
            placeholder="Enter Username"
            type="text"
            value=""
          />
        </Col>
      </FormGroup>
      <FormGroup
        bsClass="form-group"
        validationState={null}
      >
        <Col
          bsClass="col"
          componentClass={[Function]}
          sm={5}
        >
          Password
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          sm={7}
        >
          <FormControl
            bsClass="form-control"
            componentClass="input"
            onChange={[Function]}
            placeholder="Enter Password"
            type="password"
            value=""
          />
        </Col>
      </FormGroup>
    </Form>
  </Grid>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <Button
      active={false}
      autoFocus={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="btn-cancel"
      disabled={false}
      onClick={[Function]}
    >
      Cancel
    </Button>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="primary"
      disabled={true}
      onClick={[Function]}
    >
      Save
    </Button>
  </ModalFooter>
</Modal>
`;
